### PR TITLE
Fixes MAISTRA-1478: latest updates from upstream

### PR DESCRIPTION
**Pull request by dmitri-d**
Also updates to latest maistra/envoy 2.0, and updates vendored depedencies.

### CODE REVIEW

File: maistra/bazelrc-vendor [hunk 1]
```diff
@@ -51,6 +51,7 @@ build --override_repository=com_googlesource_chromium_v8=/work/maistra/vendor/co
 build --override_repository=com_googlesource_code_re2=/work/maistra/vendor/com_googlesource_code_re2
 build --override_repository=com_lightstep_tracer_cpp=/work/maistra/vendor/com_lightstep_tracer_cpp
 build --override_repository=envoy=/work/maistra/vendor/envoy
+build --override_repository=envoy_api=/work/maistra/vendor/envoy_api
```
Comment (jwendell) [inline maistra/bazelrc-vendor]: Is this right?

Comment (jwendell) [reply maistra/bazelrc-vendor]: You should probably rebuild the image (so that it gets the newer update-deps.sh) and run the command again.

Comment (dmitri-d) [reply maistra/bazelrc-vendor]: Thanks for catching this. Updated vendored dependencies.

**Comment by dmitri-d** [comment]
* Updated vendored dependencies

**Comment by dmitri-d** [comment]
* merged in latest changes from maistra-1.2 branch

**Pull request merged**